core/pythonruntime.go

```diff
@@ -16,32 +16,28 @@ func (p *Project) pythonRuntime(ctx context.Context, subpath string, gw bkgw.Cli
 	if err != nil {
 		return nil, err
 	}
+
 	workdir := "/src"
-	appdir := filepath.Join(workdir, filepath.Dir(p.ConfigPath), subpath)
-	entrypoint := fmt.Sprintf(`#!/bin/sh
-set -exu
-cd %q
-exec python -m dagger.server "$@"
-`,
-		appdir)
+
+	pipCache := llb.AddMount(
+		"/root/.cache/pip",
+		llb.Scratch(),
+		llb.AsPersistentCacheDir("pythonpipcache", llb.CacheMountShared),
+	)
 
 	return NewDirectorySt(ctx,
-		llb.Merge([]llb.State{
-			llb.Image("python:3.11-alpine", llb.WithMetaResolver(gw)).
-				Run(llb.Shlex(`apk add --no-cache file git openssh-client`)).Root(),
-			llb.Scratch().
-				File(llb.Copy(contextState, p.Directory.Dir, "/src")),
-		}).
+		llb.Image("python:3.11-alpine", llb.WithMetaResolver(gw)).
```

Comment:
general question, no need to address here, but is python generally backcompat in the 3.x range? Like will users typically be able to write code for python 3.9 and have it still work on 3.11? Or is fine grain version control a requirement?

Just wondering how soon we'll need to prioritize parameterizing this sort of thing (i.e. `"sdk": "python-3.11"). We'll absolutely need that eventually, just wondering if we can put it off or not.

Reply:
The Python SDK only supports 3.10+. I took some liberties here that use 3.11+ features since we control the runtime image, but it's not a big change to back-support 3.10 if needed. I think we can put the image customization off for a while longer. If someone needs that customization I imagine it'll be more for other reasons rather than the Python version as this is independent from the version users need to run in production.